app/src/main/res/layout/fragment_login.xml

```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ui.example.login.LoginFragment">
```

Comment:
Which is the purpose of this line?

Reply:
> Declares which activity this layout is associated with by default. This enables features in the editor or layout preview that require knowledge of the activity, such as what the layout theme should be in the preview and where to insert handlers when you make those from a quickfix 

Source: [Android Developers](https://developer.android.com/studio/write/tool-attributes#toolscontext)